File: src/main/QuizService.java
package main;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

public class QuizService {
    private Quizs quiz;

    public Quizs getQuiz() {
        return quiz;
    }

    public void setQuiz(Quizs quiz) {
        this.quiz = quiz;
    }

    public Quizs createQuiz(String quizName, Users userCreated, int duration, Set<Question> questions) {
        quiz = new Quizs();
        quiz.setName(quizName);
        quiz.setUserCreated(userCreated);
        quiz.setDuration(duration);
        quiz.setQuestions(questions);
        quiz.setCreatedAt(LocalDateTime.now());
        return quiz;
    }

    // public List< getScoreOfUserInQuiz(Users user) {
    //     for(QuizDetails quizDetail : user.getQuizDetails()) {
    //         if(quizDetail.getQuiz().getId() == quiz.getId()) {

    //         }
    //     }
    // }

    public int calculateScore(List<UserAnswers> allUserAnswer) {
        int score = 0;
        for(UserAnswers userAnswers : allUserAnswer) {
            if(checkUserAnswerInQuiz(userAnswers)) {
                score ++;
            }
        }
        return score;
    }

    public boolean checkUserAnswerInQuiz(UserAnswers userAnswer) {
        if(userAnswer == null || userAnswer.getQuestionId() == null) return false;
        for(Question quest : quiz.getQuestions()) {
            if(quest.getId() == userAnswer.getQuestionId()) {
                System.out.println(quest.getId());
                return quest.checkUserAnswer(userAnswer.getUserAnswer());
            }
        }
        return false;
    }


    public void addQuestion(Question question1) {
        quiz.addQuestion(question1);
    }


}

File: src/QuizAppCSharp/QuizApp/Service/QuizService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuizApp.Service
{
    public class QuizService
    {
        public Quiz Quiz { get; set; }

        public Quiz CreateQuiz(string quizName, User userCreated, int duration, HashSet<Question> questions)
        {
            Quiz = new Quiz();
            Quiz.Name = quizName;
            Quiz.UserCreated = userCreated;
            Quiz.Duration = duration;
            Quiz.Questions = questions;
            Quiz.CreatedAt = DateTime.Now;
            return Quiz;
        }

        public int CalculateScore(List<UserAnswer> userAnswers)
        {
            int score = 0;
            foreach (UserAnswer userAnswer in userAnswers)
            {
                if(CheckUserAnswerInQuiz(userAnswer))
                {
                    score++;
                }
            }
            return score;
        }

        public bool CheckUserAnswerInQuiz(UserAnswer userAnswer)
        {
            if (userAnswer == null) return false;
            foreach (Question question in Quiz.Questions)
            {
                if (question.Id == userAnswer.QuestionId)
                {
                    return question.CheckUserAnswer(userAnswer.AnswerOptions);
                }
            }
            return false;
        }

        public void AddQuestion(Question question)
        {
            Quiz.Questions.Add(question);
        }
    }
}
